siac/rentercmd.go

```diff
@@ -294,6 +301,35 @@ func rentercontractscmd() {
 	w.Flush()
 }
 
+// rentercontractdetailscmd is the handler for the command `siac renter contracts <id>`.
+// It lists details of a specific contract.
+func rentercontractdetailscmd(id string) {
+	var rc modules.RenterContract
+	if err := getAPI("/renter/contract/"+id, &rc); err != nil {
+		die("Could not get contract details: ", err)
+	}
+
+	// Print overview.
+	var out = ""
+	out = fmt.Sprintf("%s\n\nOVERVIEW\n========", out)
+	out = fmt.Sprintf("%s\nEnd height:\t\t%v", out, rc.EndHeight())
+	out = fmt.Sprintf("%s\nHost:\t\t\t%v", out, rc.NetAddress)
+	out = fmt.Sprintf("%s\nID:\t\t\t%v", out, rc.ID)
+
+	// File Contract.
+	out = fmt.Sprintf("%s\n\nFILE CONTRACT\n=============", out)
+	out = fmt.Sprintf("%s\nFile Size:\t\t%v", out, filesizeUnits(int64(rc.FileContract.FileSize)))
+	// TODO: finish this section.
+
+	// Spending.
+	out = fmt.Sprintf("%s\n\nSPENDING\n========", out)
+	out = fmt.Sprintf("%s\nRenter funds:\t\t%s", out, currencyUnits(rc.RenterFunds()))
+	out = fmt.Sprintf("%s\nUpload Spending:\t%v", out, currencyUnits(rc.UploadSpending))
+	// TODO: finish this section.
+
+	fmt.Println(out)
```

Comment:
Hmm, looks like the pubkey is not printing properly. `types.SiaPublicKey` has a `String` method, which should be triggered by the `%v` directive.